src/components/pinned-header-footer-card/index.cjsx

```diff
@@ -111,17 +116,44 @@ module.exports = React.createClass
     window.scroll(0, @props.buffer + @state.offset)
     @setState(shouldBeShy: true)
 
+  getHeaderHeight: ->
+    header = @refs.header?.getDOMNode()
+    headerHeight = header?.offsetHeight or 0
+
+  setOriginalContainerMargin: ->
+    container = @refs.container?.getDOMNode()
+    return unless container
+
+    @setState(containerMarginTop: window.getComputedStyle(container).marginTop) if window.getComputedStyle?
+
+  setContainerMargin: ->
+    headerHeight = @getHeaderHeight()
+    container = @refs.container?.getDOMNode()
+    return unless container
+
+    @setState(headerHeight: headerHeight)
+    if @state.pinned
+      container.style.marginTop = (headerHeight + @props.containerBuffer) + 'px'
+    else
+      container.style.marginTop = @state.containerMarginTop
+
   componentDidMount: ->
     @setOffset()
     @updatePinState(0)
+    @setOriginalContainerMargin()
+    @setContainerMargin()
+
+    window.addEventListener('resize', @setContainerMargin)
```

Comment:
Also you might consider adding a _.debounce into the flow there.  Something like:

```coffeescript
@resizeListener = _.debounce( => @setContainerMargin , 250) # season to taste 
window.addEventListener('resize',  @resizeListener)`
```

And then on unmount:

` window.removeEventListener(@resizeListener) if @resizeListener`

Some browsers are **really** aggressive on sending resize events every time the window moves a few pixels.  Without a debounce the resizing can get jerky